### PR TITLE
Add more perf dss tests

### PR DESCRIPTION
**Pull request by charleskawczynski**
Unfortunately, even though #1198 fixed some JET failures, others (added here) ended up with worse performance. The fields tested here were taken from ClimaAtmos, where this inference failure was observed. Hopefully fixing this JET test will close out #1283 for good.

### CODE REVIEW

**Comment by charleskawczynski** [comment]
This is actually seg-faulting for me locally

**Comment by charleskawczynski** [comment]
Turns out this is a DSS bug: we’re getting an OOB error (when running with check bounds, for certain types of fields). I’ll open an issue and add this test as a broken example

**Comment by charleskawczynski** [comment]
It was my mistake, I passed in the wrong buffer. That seems a bit dangerous. I'll open an issue to add some checks for this.

**Comment by charleskawczynski** [comment]
bors try

File: test/Operators/spectralelement/benchmark_kernels.jl [hunk 1]
```diff
@@ -145,3 +145,25 @@ function kernel_ntuple_field_dss!(args)
...
+end
+
+##### complicated_field dss!
+kernel_complicated_field_dss_array!(args) = kernel_copyto!(args)
+function kernel_complicated_field_dss!(args)
+    (; f_comp_buffer) = args.buffers
+    (; f_comp) = args
+    Spaces.weighted_dss_start!(f_comp, f_comp_buffer)
+    Spaces.weighted_dss_internal!(f_comp, f_comp_buffer)
+    Spaces.weighted_dss_ghost!(f_comp, f_comp_buffer)
```
Comment (sriharshakandala) [inline test/Operators/spectralelement/benchmark_kernels.jl:167]: We can just do `Spaces.weighted_dss(...)` unless there is a good reason to separate them into three function calls!

**Pull request merged**